### ISSUE TITLE
Set headers to request

### ISSUE THREAD

**Issue by julienR2** [uniqueway/redux-crud-store#32]
First thanks for this lib :) Saved me a lot of time !

I'm trying to set up "Authorization" headers ( {headers: {"Authorization":  "Token ..."}} ) to a fetch request from an action creator (fetchCollection) but I can't figure out how to do it..
- If I set it in the params of the fetchCollection, it get parsed an set up as a url params..
- The opts argument is only used to set the method from what I saw.

In the ApiClient.js it is said: 

> You can pass in config using `action.payload.fetchConfig`

But I don't managed to find out how to do this.. 

Can you give me a little help on this one ?

**Comment by devvmh** [comment]
Hi Julien, thanks for reporting this! This is a huge oversight on my part.

Let me post some thoughts, with the caveat that I haven't tested the ApiClient thoroughly so there may actually be bugs in the implementation! So definitely reply back if what I write here doesn't work.

You should be able to pass a "default" config in your ApiClient initialization currently by changing this:

```
const client = new ApiClient({ basePath: 'https://example.com/api/v1' })
```

to this:

```
const client = new ApiClient({
  basePath: 'https://example.com/api/v1',
  headers: {
    Authorization: "Token..."
  }
```

   })

This will be the new "default". 

But as you rightly point out, I added support for a "fetchConfig" parameter to be passed in the action payload, but forgot to add it to the actionCreators. I'll create a pull request ASAP for a v5.1.0 that includes this functionality. To get around this now, you could do something like this:

```
export function fetchPosts(params = {}) {
  const action = fetchCollection(MODEL, PATH, params)
  action.payload.fetchConfig = {
    headers: {
      Authorization: "Token..."
    }
  }
  return action
}
```

in all of your action creators. This is stupid, pointless boilerplate though, which is what this library is supposed to fix! So I'll try to release the fix ASAP

**Comment by devvmh** [comment]
published in 5.1.0

**Comment by julienR2** [comment]
Awesome ! Thanks for the explanation, the workaround and update !!
I was trying to do something like the workaround you posted but couldn't manage to get to the right way to do it.

I'll update my client with these changes.

Thanks again 👍
